Sources/TuistCore/Graph/GraphLoaderLinter.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+
+// MARK: - GraphLoaderLinting
+
+public protocol GraphLoaderLinting {
+    func lintWorkspace(workspace: Workspace, projects: [Project]) throws
+    func lintProject(at path: AbsolutePath, projects: [Project]) throws
+}
+
+// MARK: - GraphLoaderLinter
+
+public class GraphLoaderLinter {
```

Comment:
Nitpick: Because the linter is now executed pre-generation. I'd rename this one to `GraphLinter` and the tests file to `GraphLinterTests`.

Reply:
There is already a GraphLinter, which lint after the graph is created, do you have another suggestion?

Reply:
Would a merge make sense?

Reply:
It seems that `GraphLinter` is actually a `GeneratorGraphLinter`, similar to `CacheGraphLinter`, that means both are inside their modules `TuistGenarator` and `TuistCache`, both expecting the graph to be already created.
Maybe, since now we moved the duplicated targets name inside the `ManifestLinter` (i.e. I initially call it this way because it was potentially returning different lint errors), we could rename this to a single goal object like `CircularDependencyLinter`? 😄 WDYT?